lib/iris/tests/test_analysis.py

```diff
@@ -365,6 +365,26 @@ def test_percentile_1d(self):
                        ('analysis', 'third_quartile_foo_1d.cml'),
                        checksum=False)
 
+    def test_fast_percentile_1d(self):
+        cube = tests.stock.simple_1d()
+
+        first_quartile = cube.collapsed('foo', iris.analysis.PERCENTILE,
+                                        percent=25, percentile_method='fast')
+        np.testing.assert_array_almost_equal(first_quartile.data,
+                                             np.array([2.5], dtype=np.float32))
+        self.assertCML(first_quartile, ('analysis',
```

Comment:
This is asking @bayliffe to improve on the existing code.
Which I'm not actually against, BTW :wink: 

Ideally, I'd prefer to see a common utility routine instead of all this duplicated code...
I wouldn't insist on it, but it would make these improvements simpler + more obvious.